src/main/java/jnr/ffi/provider/jffi/DirectMemoryIO.java

```diff
@@ -186,8 +187,30 @@ public String getString(long offset) {
 
 
     public String getString(long offset, int maxLength, Charset cs) {
-        final byte[] bytes = IO.getZeroTerminatedByteArray(address() + offset, maxLength);
-        return cs.decode(ByteBuffer.wrap(bytes)).toString();
+        if(cs == StandardCharsets.UTF_8){
```

Comment:
agree that I should include other single byte terminator charsets, but CharsetEncoder.maxBytesPerChar won't be sufficient because it does not resolve to 1.0 for utf-8